go/libraries/doltcore/table/editor/index_editor.go

```diff
@@ -287,7 +288,14 @@ func RebuildIndex(ctx context.Context, tbl *doltdb.Table, indexName string, opts
 		return types.EmptyMap, fmt.Errorf("index `%s` does not exist", indexName)
 	}
 
-	rebuiltIndexData, err := rebuildIndexRowData(ctx, tbl.ValueReadWriter(), sch, tableRowData, index, opts)
+	tf := tupleFactories.Get().(*types.TupleFactory)
```

Comment:
you are done with this TF instance when the functions exits right?  I'd put it back in the pool with a defer block here.